coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -265,9 +267,15 @@ private SqlNode visitLateralViewUDTF(List<SqlNode> sqlNodes, List<SqlNode> alias
   private SqlNode visitLateralViewExplode(List<SqlNode> sqlNodes, List<SqlNode> aliasOperands,
       SqlCall tableFunctionCall, boolean isOuter) {
     final int operandCount = aliasOperands.size();
-    // array if operandCount == 3: LATERAL VIEW EXPLODE(op0) AS op1(op2)
-    // map if operandCount == 4: LATERAL VIEW EXPLODE(op0) AS op1(op2, op3)
-    checkState(operandCount == 3 || operandCount == 4,
+    // array if operandCount == 3: LATERAL VIEW EXPLODE(op0) op1 AS op2
+    // map if operandCount == 4: LATERAL VIEW EXPLODE(op0) op1 AS op2, op3
+    // if operandCount == 2: LATERAL VIEW EXPLODE(op0) op1
+    //   if op0 is a map, it implies "AS key, value" where key, value are auto-named by Hive
+    //   if op0 is an array, is implies "AS col" where col is auto-named by Hive
+    //   The logic above will be implemented as part of Calcite SqlNode validation
+    //   Note that "operandCount == 2 && isOuter" is not supported yet due to the lack of type information needed
+    //   to derive the correct IF function parameters.
+    checkState((operandCount == 2 && !isOuter) || operandCount == 3 || operandCount == 4,
```

Comment:
Is it better to throw the exception for `operandCount == 2` in the following `if (isOuter)` block? Otherwise, if user uses `operandCount == 2 && isOuter`, they will get the misleading error message because we do support `operandCount == 2 && !isOuter`

Reply:
Good catch @ljfgem !  Just fixed.  Please take another look.